addons/maintainence-agent/maintenance_mode_controller.go

```diff
@@ -54,25 +54,14 @@ func (r *MaintenanceModeReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 		return ctrl.Result{}, err
 	}
 
-	if mmode.GetDeletionTimestamp().IsZero() {
```

Comment:
Do not add the finalizer per-maturely, add it just before actual state of the system would change.

This addition here caused a MMode CR not to be deleted, when it had incorrect spec contents (specifically the ID).